### DISCUSSION TITLE
Help for non-convex polygon generate by points

### DISCUSSION THREAD

**Discussion by yulinhuyang** [boostorg/geometry#1055] [Q&A]
I used some lidar scanned boundary points to generate polygons. But these points can't fully guarantee the order. resulting in non-convexity polygon. I use bg::correct to try to fix it. It can solve some of the problems, but some still don't work. Is there any other way to generate a polygon with a compact outer frame?

![image](https://user-images.githubusercontent.com/13074332/185410049-1cae9d1a-b129-475f-8c52-ef5236343f86.png)

like this,two lines crossed.

I want get compact frame, so bg::convex_hull is not a good choice, It may generate a big frame.

**Comment by vissarion** [comment]
Concave hull could be something that might help you. See the discussion [here](https://gis.stackexchange.com/questions/1200/what-are-definition-algorithms-and-practical-solutions-for-concave-hull). Boost Geometry does not support any concave hull algorithms.
